src/Jobs/Recruit.Vacancies.Jobs/DomainEvents/Handlers/Provider/ProviderBlockedOnLegalEntityDomainEventHandler.cs

```diff
@@ -1,23 +1,50 @@
+using Esfa.Recruit.Vacancies.Client.Infrastructure.Services.ProviderRelationship;
 using Microsoft.Extensions.Logging;
 
 namespace Esfa.Recruit.Vacancies.Jobs.DomainEvents.Handlers.Provider
 {
     public class ProviderBlockedOnLegalEntityDomainEventHandler : DomainEventHandler, IDomainEventHandler<ProviderBlockedOnLegalEntityEvent>
     {
-        ILogger<ProviderBlockedOnLegalEntityDomainEventHandler> _logger;
-        public ProviderBlockedOnLegalEntityDomainEventHandler(ILogger<ProviderBlockedOnLegalEntityDomainEventHandler> logger) : base(logger)
+        private readonly IEmployerAccountProvider _employerAccountProvider;
+        private readonly IProviderRelationshipsService _providerRelationshipsService;
+        private readonly ILogger<ProviderBlockedOnLegalEntityDomainEventHandler> _logger;
+        public ProviderBlockedOnLegalEntityDomainEventHandler(
+            IEmployerAccountProvider employerAccountProvider,
+            IProviderRelationshipsService providerRelationshipsService,
+            ILogger<ProviderBlockedOnLegalEntityDomainEventHandler> logger) : base(logger)
         {
             _logger = logger;
+            _employerAccountProvider = employerAccountProvider;
+            _providerRelationshipsService = providerRelationshipsService;
         }
 
-        public Task HandleAsync(string eventPayload)
+        public async Task HandleAsync(string eventPayload)
         {
             var eventData = DeserializeEvent<ProviderBlockedOnLegalEntityEvent>(eventPayload);
-            //TODO 
-            // call provider permissions api to revoke permission for each legal entity
-            return Task.CompletedTask;
+            
+            _logger.LogInformation($"Attempting to revoke provider {eventData.Ukprn} permission on account {eventData.EmployerAccountId} for legal entity {eventData.LegalEntityId}.");
+
+            var accountLegalEntityPublicHashedId = await GetAccountLegalEntityPublicHashId(eventData);
+            if(string.IsNullOrWhiteSpace(accountLegalEntityPublicHashedId))
+            {
+                _logger.LogError($"The legal entity {eventData.LegalEntityId} for the account {eventData.EmployerAccountId} was not found.");
+                return;
+            }
+
+            await _providerRelationshipsService.RevokeProviderPermissionToRecruitAsync(eventData.Ukprn, accountLegalEntityPublicHashedId);
+
+            _logger.LogInformation($"Successfully revoked provider {eventData.Ukprn} permission on account {eventData.EmployerAccountId} for legal entity {eventData.LegalEntityId}.");
+        }
+
+        private async Task<string> GetAccountLegalEntityPublicHashId(ProviderBlockedOnLegalEntityEvent eventData)
+        {
+            var legalEntities = await _employerAccountProvider.GetLegalEntitiesConnectedToAccountAsync(eventData.EmployerAccountId);
+
+            return legalEntities.FirstOrDefault(l => l.LegalEntityId == eventData.LegalEntityId).AccountLegalEntityPublicHashedId;
```

Comment:
This is done deliberately. There is an assumption that this should always work. If in case there is a failure, I want it to go bang right here so we can find it in the stack trace. Returning null means I will have to check for nulls else where and then handle that situation which may not occur ever.